numpy/core/src/multiarray/item_selection.c

```diff
@@ -1749,6 +1750,275 @@ PyArray_LexSort(PyObject *sort_keys, int axis)
+            || (PyArray_STRIDES(mps[j])[axis] != (npy_intp)PyArray_DESCR(mps[j])->elsize);
+        if (PyArray_DESCR(mps[j])->elsize > maxelsize) {
+            maxelsize = PyArray_DESCR(mps[j])->elsize;
+        }
+    }
+    assert(N > 0);  /* Guaranteed and assumed by indbuffer */
+    char *indbuffer = PyDataMem_NEW(N * sizeof(npy_intp));
+    if (indbuffer == NULL) {
+       PyErr_NoMemory();
+       goto fail;
+    }
+    if (needcopy) {
+        char *valbuffer;
+        // Create a buffer of size of the current 1D slice i.e N
+        npy_intp valbufsize = N * maxelsize;
+        if (NPY_UNLIKELY(valbufsize) == 0) {
+            valbufsize = 1;  /* Ensure allocation is not empty */
+        }
+
+        valbuffer = PyDataMem_NEW(valbufsize);
+        if (valbuffer == NULL) {
+            PyDataMem_FREE(indbuffer);
+            PyErr_NoMemory();
+            goto fail;
+        }
+        while (size--) {
+            iptr = (npy_intp *)indbuffer;
+            for (i = 0; i < N; i++) {
+                *iptr++ = i;
+            }
+            for (j = 0; j < n; j++) {
+                int rcode;
+                elsize = PyArray_DESCR(mps[j])->elsize;
+                astride = PyArray_STRIDES(mps[j])[axis];
+                argsort = PyArray_DESCR(mps[j])->f->argsort[NPY_STABLESORT];
+                if(argsort == NULL) {
+                    argsort = npy_atimsort;
+                }
+                _unaligned_strided_byte_copy(valbuffer, (npy_intp) elsize,
+                                             its[j]->dataptr, astride, N, elsize);
+                if (PyArray_ISBYTESWAPPED(mps[j])) {
+                    _strided_byte_swap(valbuffer, (npy_intp) elsize, N, elsize);
+                }
+                rcode = argsort(valbuffer, (npy_intp *)indbuffer, N, mps[j]);
+                if (rcode < 0 || (PyDataType_REFCHK(PyArray_DESCR(mps[j]))
+                            && PyErr_Occurred())) {
+                    PyDataMem_FREE(valbuffer);
+                    PyDataMem_FREE(indbuffer);
+                    goto fail;
+                }
+                PyArray_ITER_NEXT(its[j]);
+            }
+            char *slicebuffer = PyDataMem_NEW(N*self_elsize);
+            _unaligned_strided_byte_copy(slicebuffer, self_elsize, sit->dataptr,
+                                         self_stride, N, self_elsize);
+
+            iptr = (npy_intp *)indbuffer;
+            for(i = 0; i < N; i++){
+                int index = *iptr;
+                char *startOld = slicebuffer + (index * self_elsize);
+                memmove(sit->dataptr + (i*self_stride),
+                            startOld, self_elsize);
+                iptr++;
+            }
+            PyArray_ITER_NEXT(sit);
+            PyDataMem_FREE(slicebuffer);
+        }
+        PyDataMem_FREE(valbuffer);
+        PyDataMem_FREE(indbuffer);
+    }
+    else {
+        while (size--) {
+            iptr = (npy_intp *)indbuffer;
```

Comment:
Better to reduce the scope of this variable
```suggestion
            npy_intp *iptr = (npy_intp *)indbuffer;
```

Reply:
Would recommend doing this wherever possible.